src/modules/tasksClass.js

```diff
@@ -74,17 +73,23 @@ export class TasksClass {
     localStorage.setItem('taskList', JSON.stringify(this.taskCollection));
   } // edit task
 
-  taskCompleted = (theElIndex, taskBoxChecked) => {
+  taskCompleted = (theElIndex, taskBoxValue) => {
     const tC = this.taskCollection;
-    tC[theElIndex].taskCompletion = taskBoxChecked;
+    tC[theElIndex].taskCompletion = taskBoxValue;
     localStorage.setItem('taskList', JSON.stringify(this.taskCollection));
   } // task is marked complete
 
-  clearCompletedTask = (clearChecks) => {
-    let tC = this.taskCollection;
-    tC = tC.filter((value) => value.taskCompletion === true);
-    tC.taskCompletion = clearChecks;
-    localStorage.setItem('taskList', JSON.stringify(this.taskCollection));
+  clearCompletedTask = (theEl, theElIndex) => {
+    const tC = this.taskCollection;
+    tC.filter((task) => task.taskCompletion === false);
+    tC.splice(theElIndex, 1);
+    // rearrange by sorting using their index
+    tC.sort((task1, task2) => task1.taskIndex - task2.taskIndex);
+    // reassign task index by iterations
+    tC.forEach((taskItem, taskItemIndex) => {
+      taskItem.taskIndex = taskItemIndex + 1;
+    });
+    localStorage.setItem('taskList', JSON.stringify(tC));
```

Comment:
Just as the previous reviewer has requested, your `clear completed` functionality doesn't work properly😉. just as it is required when the button is clicked, the tasks that have been marked completed should be erased from the local storage and the UI too😉. Kindly see the video below to see the current behavior of the app😉 and make the necessary change as required👍..


https://github.com/ometman/to-do-list/assets/64643995/4bb8d788-1670-4965-b19f-a6045621cb3d

Reply:
Working fine locally. But will check it. Thanks